faiss_vector_posting.go

```diff
@@ -417,20 +417,45 @@ func (sb *SegmentBase) InterpretVectorIndex(field string, requiresFiltering bool
 					// Getting the vector IDs corresponding to the eligible
 					// doc IDs.
 					eligibleVecIDsBitmap := roaring.NewBitmap()
+					vecIDsUint32 := make([]uint32, 0)
```

Comment:
Hmm, couple things ..

- does docVecIDMap need to of type map[uint32][]int64 and not map[uint32][]uint32?
- maybe better to not allocate here and do this instead .. (because `append` below in your code will become an alloc too)

```
eligibleVecIDsBitmap := roaring.NewBitmap()
for _, eligibleDocID := range eligibleDocIDs {
    vecIDs := docVecIDMap[uint32(eligibleDocID)]
    if len(vecIDs) > 0 {
        vecIDsUint32 := make([]uint32, len(vecIDs)
        for _, vecID := range vecIDs {
             vecIDsUint32 = append(vecIDsUint32, uint32(vecID))
        }
        eligibleVecIDsBitmap.AddMany(vecIDsUint32)
}
```

Reply:
About the first question, it's int64 since the vector IDs are of type int64.
I discussed this with @Thejas-bhat yesterday and he said that the rationale behind originally picking int64, over int32, was to minimise the chance of hash collisions due to overflow when computing the hash.

Reply:
@Thejas-bhat pls let me know if I got that right.

Reply:
yeah, makes sense to me 

another point here is that faiss accepts the ids as int64 type from what i see

https://github.com/blevesearch/faiss/blob/b747c55a93a9627039c34d44b081f375dca94e57/faiss/MetricType.h#L39
https://github.com/blevesearch/faiss/blob/b747c55a93a9627039c34d44b081f375dca94e57/faiss/Index.h#L116

Reply:
About the second point, I think your approach might have the downside of adding multiple batches to the eligibleVecIDsBitmap, instead of the current approach which adds a single one.

Reply:
Roaring is a much more efficient data structure, with the current approach - we'd be building one large uint32 array as opposed to multiple smaller ones - for GO GC, would the former be better or latter?

Reply:
Per clarification from Aditi - it seems the []int64 of vector IDs in the docVecIDMap is usually of length 1, except in the event of nested vectors, in which event the current implementation should be good.

@metonymic-smokey would you add some commentary around this here, also perhaps a couple examples showing how the docVecIDMap looks typically with and without nested vectors in those comments would be greatly helpful 👍🏼